lib/maze/request_list.rb

```diff
@@ -64,5 +64,19 @@ def clear
       @current = 0
       @count = 0
     end
+
+    # Sorts the first `count` elements of the list by the Bugsnag-Sent-At header, if present in all of those elements
+    def sort_by_sent_at!(count)
+      return unless count > 1
+
+      header = 'Bugsnag-Sent-At'
+      sub_list = @requests[@current...@current + count]
+
+      return unless sub_list.all? { |r| !r[:request][header].nil? }
```

Comment:
nit: maybe this is a bit easier to read as "return if any element of the list has a `nil` sent-at header"

```suggestion
      return if sub_list.any? { |r| r[:request][header].nil? }
```

Reply:
Yes, much better!